### PR TITLE
Remove Writeable#readFrom

### PR DESCRIPTION
**Pull request by nik9000**
It is always better to call a static read method or a constructor that
takes StreamInput.

Relates to #17085

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/common/io/stream/Writeable.java [hunk 1]
```diff
@@ -39,16 +39,6 @@
     void writeTo(StreamOutput out) throws IOException;
 
     /**
-     * Read this object from a stream. Use a {@link Writeable.Reader} instead. This lives on for backwards compatibility but should be
-     * removed before 5.0.0GA. It is not deprecated because Diffable extends this interface and it shouldn't be deprecated there.
-     */
-    default T readFrom(StreamInput in) throws IOException {
```
Comment (cbuescher) [inline core/src/main/java/org/elasticsearch/common/io/stream/Writeable.java]: finally! its gone!

**Comment by cbuescher** [comment]
LGTM

**Comment by nik9000** [comment]
Thank so much for all the reviews @cbuescher !

File: core/src/main/java/org/elasticsearch/common/io/stream/Writeable.java [hunk 2]
```diff
@@ -39,16 +39,6 @@
...
-        // NORELEASE remove before 5.0.0GA
-        throw new UnsupportedOperationException(
-                "Prefer calling a constructor or static method that takes a StreamInput to calling readFrom.");
-    }
-
-    /**
```
Comment (javanna) [inline core/src/main/java/org/elasticsearch/common/io/stream/Writeable.java]: can we remove the type T from this class and the corresponding TODO?

Comment (nik9000) [inline core/src/main/java/org/elasticsearch/common/io/stream/Writeable.java]: Happening right now!

**Pull request merged**